ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/stratum/Stratum1Protocol.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.blockcreation.stratum;
+
+import static org.apache.logging.log4j.LogManager.getLogger;
+
+import org.hyperledger.besu.ethereum.mainnet.DirectAcyclicGraphSeed;
+import org.hyperledger.besu.ethereum.mainnet.EthHashSolverInputs;
+import org.hyperledger.besu.util.bytes.BytesValue;
+import org.hyperledger.besu.util.bytes.BytesValues;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Implementation of the stratum+tcp protocol.
+ *
+ * <p>This protocol allows miners to submit nonces over a persistent TCP connection.
+ */
+public class Stratum1Protocol implements StratumProtocol {
+
+  private EthHashSolverInputs currentInput;
+  private Function<Long, Boolean> submitCallback;
+  private Supplier<String> jobIdSupplier =
+      () -> {
+        BytesValue timeValue = BytesValues.toMinimalBytes(Instant.now().toEpochMilli());
+        return timeValue.slice(timeValue.size() - 4, 4).toUnprefixedString();
+      };
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Style wise we have been enumerating ignored properties and also annotating the beans properties with @JsonProperty, even though it is redundant.

Reply:
OK that's fixed. Moved to a JsonCreator annotation as well.